src/server/process_request.c

```diff
@@ -1015,7 +1015,7 @@ dispatch_request(int sfds, struct batch_request *request)
 			break;
 
 		case PBS_BATCH_StatusNode:
-			if (set_to_non_blocking(conn) == -1) {
+			if (prot != PROT_TPP && set_to_non_blocking(conn) == -1) {
```

Comment:
not a comment for this time, but i could not put a comment on code you had committed in an earlier PR, hence hanging it here. the ptask structure is used only by the PBS_BATCH_ServerReady handler req_stat_svr_ready(). So why not set it inside req_stat_svr_ready() and pass only request to it. It seems contrived that we assign request to wt_parm of ptask and then pass a pointer to it. Why?

Reply:
req_stat_svr_ready replies back only if the server is in a consistent state. Otherwise, it will leave a work task with the request in it which will be converted to immediate when all acks are received. So req_stat_svr_ready needs a wrapper function that does accept request structure and set it to the ptask. I choose to do it in the caller as there is no other thing that wrapper can do. I'll leave a comment indicating this.

Reply:
Well, the actual answer is that req_stat_svr_ready() is set as a work task and hence you wanted the parameter to be work_task * and not request *. I see that now and it is fine.